enterprise/backend/src/metabase_enterprise/serialization/api.clj

```diff
@@ -0,0 +1,168 @@
+;;; Request callbacks
+
+(defn- ba-copy [f]
+  (let [baos (ByteArrayOutputStream.)]
+    (io/copy f baos)
+    (.toByteArray baos)))
+
+(defn- on-response! [data callback]
+  (reify
+    ;; Real HTTP requests and mt/user-real-request go here
+    ring.protocols/StreamableResponseBody
+    (write-body-to-stream [_ response out]
+      (ring.protocols/write-body-to-stream data response out)
+      (future (callback)))
+
+    ;; mt/user-http-request goes here
+    clojure.java.io.IOFactory
+    (make-input-stream [_ _]
+      (let [res (io/input-stream (if (instance? File data)
+                                   (ba-copy data)
+                                   data))]
+        (callback)
+        res))))
+
+;;; Logic
+
+(defn- serialize&pack ^File [{:keys [dirname] :as opts}]
+  (let [dirname (or dirname
+                    (format "%s-%s"
+                            (u/slugify (public-settings/site-name))
+                            (u.date/format "YYYY-MM-dd_HH-mm" (t/local-date-time))))
+        path     (io/file parent-dir dirname)
+        dst      (io/file (str (.getPath path) ".tar.gz"))
+        log-file (io/file path "export.log")]
+    (with-open [_logger (logger/for-ns 'metabase-enterprise.serialization log-file)]
+      (try                              ; try/catch inside logging to log errors
+        (serdes/with-cache
+          (-> (extract/extract opts)
+              (storage/store! path)))
+        ;; not removing storage immediately to save some time before response
+        (u.compress/tgz path dst)
+        (catch Exception e
+          (log/error e "Error during serialization"))))
+    {:archive  (when (.exists dst)
+                 dst)
+     :log-file (when (.exists log-file)
+                 log-file)
+     :callback (fn []
+                 (when (.exists path)
+                   (run! io/delete-file (reverse (file-seq path))))
+                 (when (.exists dst)
+                   (io/delete-file dst)))}))
+
+(defn- unpack&import [^File file & [size]]
+  (let [dst      (io/file parent-dir (u.random/random-name))
+        log-file (io/file dst "import.log")]
+    (with-open [_logger (logger/for-ns 'metabase-enterprise.serialization log-file)]
+      (try                              ; try/catch inside logging to log errors
+        (log/infof "Serdes import, size %s" size)
+        (let [path (u.compress/untgz file dst)]
+          (serdes/with-cache
+            (-> (v2.ingest/ingest-yaml (.getPath (io/file dst path)))
+                (v2.load/load-metabase! {:abort-on-error true}))))
+        (catch Exception e
+          (log/error e "Error during serialization"))))
+    {:log-file log-file
+     :callback #(when (.exists dst)
+                  (run! io/delete-file (reverse (file-seq dst))))}))
+
+;;; HTTP API
+
+(defendpoint2 POST "/export"
+  "Serialize and retrieve Metabase instance.
+
+  Parameters:
+  - `dirname`: str, name of directory and archive file (default: `<instance-name>-<YYYY-MM-dd_HH-mm>`)
```

Comment:
It seems like docstrings can be generated, which would keep it up to date and complete.

 Have you considered generating this by adding attributes to the malli schemas in the map below? Something like:

```
dirname [:maybe {:mb/doc "name of directory and archive file" 
                 :default "<instance-name>-<YYYY-MM-dd_HH-mm>"} 
         :string]
```